lib/linguist/heuristics.rb

```diff
@@ -73,7 +73,7 @@ def call(data)
       if (/@(interface|class|protocol|property|end|synchronised|selector|implementation)\b/.match(data))
```

Comment:
ok @arfon, having given it some thought, the revised regex needs to be:

/^[ \t]*@(interface|class|protocol|property|end|synchronised|selector|implementation)\b/